src/main/java/nl/tudelft/bep/deeplearning/clustering/NearestNeighbors.java

```diff
@@ -0,0 +1,182 @@
+package nl.tudelft.bep.deeplearning.clustering;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+public class NearestNeighbors {
```

Comment:
This is a utility class. Requires a final class modifier and private constructor.